agent/router/router.go

```diff
@@ -116,6 +116,9 @@ func (r *Router) Shutdown() {
 
 // AddArea registers a new network area with the router.
 func (r *Router) AddArea(areaID types.AreaID, cluster RouterSerfCluster, pinger Pinger, useTLS bool) error {
+	if cluster == nil {
+		return nil
+	}
```

Comment:
Is returning an error appropriate here given that we guard the call in `NewServerLogger` when cluster is nil anyway? Maybe there are other callers that would fail if it were an error.